### PR TITLE
[mathgl] Update, fix cmake config, dependencies, linkage, usage

### LINKED ISSUE #22264: CMake not able to find MathGL
**Describe the bug**
find_package does not find the MathGL library. I have installed the package mathgl:x64-windows. I have other packages installed with vcpkg and none of them cause any trouble.

**Environment**
- OS: Windows 10
- Compiler: MSVC 19.29.30138
-CMake version 3.22.0 
-Commit 95d4d067f05b36225b292bf1c3156a7a20c913c0 of the vcpkg repo

**To Reproduce**
1. ./vcpkg.exe install mathgl:x64-windows
2. Try to find the package as instructed in vcpkg:
In a CMakeLists.txt file:
find_package(mathgl CONFIG REQUIRED)
2. cmake -DCMAKE_TOOLCHAIN_FILE="[path]" -DVCPKG_TARGET_TRIPLET="x64-windows" ..

**Expected behavior**
CMake should be able to find the package. On my windows 10 machine, it does not. I also have other packages installed with vcpkg (such as eigen3 and gsl) which CMake is able to find without causing any problem.

**Failure logs** 
 (_find_package):
  Could not find a package configuration file provided by "mathgl" with any
  of the following names:

    mathglConfig.cmake
    mathgl-config.cmake

  Add the installation prefix of "mathgl" to CMAKE_PREFIX_PATH or set
  "mathgl_DIR" to a directory containing one of the above files.  If "mathgl"
  provides a separate development package or SDK, be sure it has been
  installed.

**Additional context**

### PR DESCRIPTION
**Pull request by dg0yt**
#### What does your PR fix?
- Updates to 8.0.1.
- Fixes cmake config. Adds usage document. Resolves #22264, #28285.
- Enables osx by using pthreads instead of openmp.
- Fixes dependencies and linkage problems. In particular, 
  - Uses proper `find_package(<Pkg>)` instead of `include(Find<Pkg>)` which did circumvent vcpkg wrappers.
  - De-vendors getopt.
  - Disables gettext/libintl (would need more work at least for osx).
  - Enables glut on osx by using system libs instead of freeglut.
- Move platform support state from ci baseline to manifest
  - uwp builds will fail at least on tools because there is no getopt implementation.
- Adds a test port which requires all features.
- Controls building of examples.
  The examples don't get installed, but they are important to validate all features and dependencies in the test port. 

Remarks:
- The installed cmake files raised some questions, but I didn't find upstream documentation on canonical cmake usage.
- The exported cmake config didn't receive the proper fixup for tool location because it the `install` commands were used with absolute paths. (`vcpkg_cmake_config_fixup` doesn't handle this case.)
- Upstream builds both static and shared libs, and uses them in a mixed way for executables. This PR excludes the undesired libs from the `all` target, and ensures that executables are linked to the desired lib (via `${linktype}`). In this way, building the undesired lib is really avoided, and the desired lib is validated with regard to dll import/export markup and to required link libraries.
- Some patches could be upstreamed now.

### CODE REVIEW

**Comment by dg0yt** [comment]
TODO: Cleanup, trim patching.
There is a choice for linkage of included executables: Upstream always uses the static lib, so it must be built even if it is not installed. The current patch actually switches the linkage of the executables. This was a useful to discover linkage problems, but it is not really a "minimal" patch.

**Review by dg0yt — COMMENTED** [review]
For CI testing, this still includes the separate PRs #28558 and #28575.
When reviewing, ignore the changes to fltk and freeglut.

**Comment by dg0yt** [comment]
I will resolve the merge conflict after https://github.com/microsoft/vcpkg/pull/28575 is also merged.

File: ports/mathgl/dependencies.patch [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+diff --git a/scripts/MathGLConfig.cmake.in b/scripts/MathGLConfig.cmake.in
+index 78bad89..0c2e5ed 100644
+--- a/scripts/MathGLConfig.cmake.in
++++ b/scripts/MathGLConfig.cmake.in
+@@ -25,6 +25,12 @@ endif()
+ if (MathGL2_HAVE_OPENMP STRGREATER "")
+   find_dependency(OpenMP)
+ endif()
++if("@enable-arma@")
++  find_package(Armadillo CONFIG)
```
Comment (LilyWangLL) [inline ports/mathgl/dependencies.patch:111]:
```suggestion
+  find_dependency(Armadillo CONFIG)
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/28522

File: ports/mathgl/dependencies.patch (hunk 1 above)
Comment (dg0yt) [reply ports/mathgl/dependencies.patch]: Done.

**Comment by dg0yt** [comment]
> Depends on #28522

Resolved.

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**